### PR TITLE
Feature/armor

### PR DESCRIPTION
**Pull request by rikka0w0**
Implements callbacks in IForgeItem:
1. onArmorTick
2. getArmorTexture
3. getArmorModel

### CODE REVIEW

File: patchwork-extensions-item/src/main/java/net/patchworkmc/mixin/extensions/item/client/MixinArmorFeatureRenderer.java [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+	 * More generic ForgeHook version of the above function, it allows for Items to
+	 * have more control over what texture they provide.
+	 *
+	 * @param entity Entity wearing the armor
+	 * @param stack  ItemStack for the armor
+	 * @param slot   Slot ID that the item is in
+	 * @param type   Subtype, can be null or "overlay"
+	 * @return ResourceLocation pointing at the armor's texture
+	 */
+	public Identifier getArmorResource(Entity entity, ItemStack stack, EquipmentSlot slot, @Nullable String type) {
```
[the remark this answers is not in the document]
Comment (rikka0w0) [reply patchwork-extensions-item/src/main/java/net/patchworkmc/mixin/extensions/item/client/MixinArmorFeatureRenderer.java:160]: It is a forge callback, so I'm going to add it to an interface

File: patchwork-extensions-item/src/main/java/net/patchworkmc/mixin/extensions/item/client/MixinArmorFeatureRenderer.java [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+			texture = texture.substring(idx + 1);
+		}
+
+		String s1 = String.format("%s:textures/models/armor/%s_layer_%d%s.png", domain, texture, (isLegs(slot) ? 2 : 1),
+				type == null ? "" : String.format("_%s", type));
+
+		s1 = PatchworkArmorItemHandler.getArmorTexture(entity, stack, s1, slot, type);
+		Identifier resourcelocation = (Identifier) ARMOR_TEXTURE_CACHE.get(s1);
+
+		if (resourcelocation == null) {
```
[the remark this answers is not in the document]
Comment (rikka0w0) [reply patchwork-extensions-item/src/main/java/net/patchworkmc/mixin/extensions/item/client/MixinArmorFeatureRenderer.java]: removed

**Comment by rikka0w0** [comment]
checkLicense failed due to
```
* What went wrong:
Execution failed for task ':patchwork-networking-messages:checkLicenseMain'.
> License violations were found: src/main/java/net/patchworkmc/impl/networking/PatchworkPlayNetworkingMessagesClient.java
```

**Comment by TheGlitch76** [comment]
Merging with master should fix the license check.

**Comment by TheGlitch76** [comment]
Merge conflicts

**Pull request merged**